de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java

```diff
@@ -209,6 +233,53 @@ protected SWTBotTreeItem openEditor(SWTBotTreeItem item) {
 	 * @param diagramEditor Graphiti diagram editor which the tree item is beeing dragged onto
 	 */
 	protected void dragTreeItemToDiagramEditor(SWTBotTreeItem item, SWTBotGefEditor diagramEditor) {
+		item.dragAndDrop(getCanvasForDiagramEditor(diagramEditor));
+		waitForEditingDomainAndUiThread();
+	}
+	
+	/**
+	 * Returns the absolute rectangular bounds for a SWTBotGefEditpart element
+	 * @param SWTBotEditPart edit part to get bounds for
+	 * @return Rectangle bounds for SWTBotGefEditPart
+	 */
+	protected Rectangle getBoundsForEditPart(SWTBotGefEditPart swtBotEditPart) {	
+		IFigure figure = ((GraphicalEditPart) swtBotEditPart.part()).getFigure();
+		Rectangle bounds = figure.getBounds().getCopy();
+		figure.translateToAbsolute(bounds);		
+		return bounds;
+	}
+	
+	/**
+	 * We can only drop tree items to canvas elements. SWTBotGefViewer's canvas element is not 
+	 * capable of beeing accessed directly from external classes though. Further more SWTBotGefViewer's 
+	 * canvas member gets populated during runtime only. Hence we use reflections to get a hold on it here.
+	 * 
+	 * @param item Tree item that is beeing dragged
+	 * @param diagramEditor Graphiti diagram editor which the tree item is beeing dragged onto
+	 */
+	protected void dragTreeItemToDiagramEditor(SWTBotTreeItem item, SWTBotGefEditor diagramEditor, int x, int y) {
+		item.dragAndDrop(getCanvasForDiagramEditor(diagramEditor), new Point(x, y));
+		waitForEditingDomainAndUiThread();
+	}
+	
+	/**
+	 * Drags a tree item onto a specific edit part element within a specified graphiti diagram editor 
+	 * @param item Tree item that is beeing dragged
+	 * @param diagramEditor Graphiti diagram editor in which the editPart is present
+	 * @param editPart editPart which the tree item is beeing dragged onto
+	 */
+	protected void dragTreeItemOnToEditPart(SWTBotTreeItem item, SWTBotGefEditor diagramEditor, SWTBotGefEditPart editPart) {

```

Comment:
Would it maybe be possible to infer the diagramEditor from the editPart by going upwards in the parent() method?
I think this could simplify the API and reduce the number of methods.

Reply:
you can only go up till you reach the root editPart, but you can't get the diagramEditor as it is a GefViewer element that editParts have no reference to.

Reply:
Oh okay, so there is no way to infer the diagram editor from the part?

Reply:
Unfortunately, I don't see how